crates/typst-kit/src/project.rs

```diff
@@ -0,0 +1,58 @@
+//! Identify and manage Typst projects.
```

Comment:
I'm not sure whether this should really exist in `typst-kit` at this moment. Since the `typst.toml` file is only used in packages, there simply isn't really a way to properly identify projects, but this makes it seem like there is.

Reply:
I figured once git packages (where normal consumer projects will have a `typst.toml`) or other ways to identify a project are added they could be added here and people get the same API.

I kept it as simple as it is here, because in my downstream implementation I have optional fallbacks to look for `src/main.typ`, `src/lib.typ`, `main.typ` and various tooling configuration like `typstfmt.toml`, which obviously didn't belong here.

I'll remove it then, doesn't make much sense yet.